src/components/login/Login.js

```diff
@@ -0,0 +1,146 @@
+import { useEffect, useReducer } from "react";
+import styles from "./Login.module.css";
+import { useAuth } from "../../context";
+import { useNavigate, useLocation } from "react-router-dom";
+import { loginUserApi } from "../../fakeAuthApi";
+
+const initialForFieldsState = {
+  email: "",
+  password: "",
+  loginError: "",
+};
+
+const formFieldsReducer = (state, action) => {
+  switch (action.type) {
+    case "EMAIL":
+      return { ...state, email: action.payload.email };
+
+    case "PASSWORD":
+      return { ...state, password: action.payload.password };
+
+    case "LOGIN_ERROR":
+      return { ...state, loginError: action.payload.loginError };
+
+      default:
+        return state;
+  }
+};
+
+export const Login = () => {
+  const [{ email, password, loginError }, dispatchFormFields] = useReducer(
+    formFieldsReducer,
+    initialForFieldsState
+  );
+
+  const { state } = useLocation();
+  const navigate = useNavigate();
+
+  const { isUserLoggedIn,dispatchAuth } = useAuth();
+
+  useEffect(()=>{
+    isUserLoggedIn && navigate(state?.from ? `${state.from}` : "/");
+  },[])
+
+
+
+  const emailRegex = /^[a-zA-Z0-9+_.-]+@[a-zA-Z0-9.-]+$/;
+  let emailError = "";
+  let passwordError = "";
+
+  if (email !== "" && !emailRegex.test(email)) {
+    emailError = "please enter valid email";
+  } else {
+    emailError = "";
+  }
+
+  const loginUser = async (enteredEmail, enteredPassword) => {
+    try {
+      dispatchFormFields({ type: "LOGIN_ERROR", payload: { loginError: "" } });
+      const response = await loginUserApi(enteredEmail, enteredPassword);
+      if (response.status === 200) {
+        const { name, email, password } = response.data;
+        dispatchAuth({
+          type: "SET_CREDENTIALS",
+          payload: { name, email, password },
+        });
+        dispatchAuth({ type: "LOGIN_USER" });
+        localStorage?.setItem(
+          "login",
+          JSON.stringify({
+            isUserLoggedIn: true,
+            name: name,
+            email: email
+          })
+        );
+        navigate(state?.from ? state.from : "/");
+      }
+    } catch (error) {
+      console.log("Error while logging in", error);
+      dispatchFormFields({
+        type: "LOGIN_ERROR",
```

Comment:
Just a suggestion: These actions types can be moved to a separate file and export from there if in future if it  is getting changing then changing in that file would do the trick.

Reply:
sure will do that